tensorflow/java/src/main/java/org/tensorflow/op/OperationHelper.java

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.op;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.tensorflow.Operation;
+import org.tensorflow.OperationBuilder;
+import org.tensorflow.Output;
+
+/**
+ * A helper for wrapping a new operation.
+ *
+ * <p>It provides utilities for wrappers to build a single operation and collect its output(s)
+ * afterward.
+ *
+ * <pre>{@code
+ * public final class Unique extends AbstractSingleOp {
+ *    ...
+ *    public static Unique create(Scope s, Input x) {
+ *        OperationHelper unique = OperationHelper.create(s, "Unique");
```

Comment:
The vast majority of operations will be simpler ones (i.e, not involve lists). Furthermore, even for the operations that produce lists of tensors as named outputs, it appears the contents of `nextOutputList()` is simple enough that unrolling it will make the custom `Op`s code easier to follow too.

I understand that this is judgement on aesthetics and thus there is no right or wrong, just differing opinions.  I'm of the opinion that optimizing for readability of the code for the common case is worth the few extra lines for list-producing operation wrappers. Furthermore, if a helper is really needed, perhaps it is better placed as a method on the `Operation` class (e.g., `Operation.outputListStart(String name)` or something, that returns the first output index at which the named output's elements begin), instead of having this helper class that is essentially wrapping an `Operation` object and a counter.

Also, the fact that we'll have a public class with a name `OperationHelper` makes me a bit uneasy. Something about the name isn't descriptive enough, which makes me feel that it shouldn't be a public API class.